### PR TITLE
Add support for Klein instance copy

### PR DESCRIPTION
**Pull request by joac**

### CODE REVIEW

**Comment by joac** [comment]
If more tests are needed, please let me know.

**Comment by posita** [comment]
@joac, please forgive my ignorance, but can you provide a use case where creating a deep copy resolves an issue that is otherwise difficult?

**Comment by joac** [comment]
@posita In the case you want to reuse an existing router:

``` python
class BaseApi(object):
    router = Klein()

    @router.route('/')
    def index(self, request):
         return "foo"

class ChildApi(BaseApi):
    router = copy.copy(BaseApi.router)

    @router.route('/bar')
    def bar(self, request):
        return 'bar'

class OtherChildApi(BaseApi):
    router = copy.copy(BaseApi.router)
    
    @router.route('/bar')
    def other_bar(self, request):
        return "This is other bar"
```

File: klein/test/test_app.py [hunk 1]
```diff
@@ -279,3 +280,41 @@ def test_resource(self, mock_kr):
 
         mock_kr.assert_called_with(app)
         self.assertEqual(mock_kr.return_value, resource)
+
+    def test_copy(self):
```
Comment (glyph) [inline klein/test/test_app.py]: Could you possibly add docstrings for these test methods?  https://jml.io/pages/test-docstrings.html is a good guideline for how.

**Comment by glyph** [comment]
@joac - the feedback on this branch is pretty minor - can you merge in master and do the couple of cleanups?

File: klein/test/test_app.py (hunk 1 above)
Comment (joac) [inline klein/test/test_app.py]: Ok will add

**Comment by glyph** [comment]
Hey @joac - ping?  Any chance you could complete those cleanups so we can land this?

**Comment by joac** [comment]
@glyph sorry for the long delay, I fixed the conflicts and added the test docstrings. Let me know if there is some issue

**Comment by glyph** [comment]
@joac - Bump - still waiting for a passing CI build and coverage metrics :)

**Comment by wsanchez** [comment]
Please update from trunk.  Apparently there's an "allow other contributors to push to this branch" button you can enable that would have allowed me to do that for you.

I think you can also fix the coverage errors by using lambdas for your error handlers instead of defining methods that aren't called.

**Comment by joac** [comment]
@wsanchez I added that permission, sorry for the delay!

**Comment by glyph** [comment]
Checking in on this to update the branch, I notice some conflicts have crept in :(

**Comment by glyph** [comment]
I think I understand the need for klein instance cloning a bit better when I last looked at this.  If I can find the time, I'll address the conflicts and merge it myself; but it probably needs some documentation to explain how to use it.

**Comment by joac** [comment]
Sorry, I didn't keep it updated

**Comment by glyph** [comment]
@joac No worries.  If anything we should be apologizing for the delay :)

**Comment by glyph** [comment]
This should probably land, although some vagaries about how Git models file moves seem to have left this in an unfortunate state with respect to its merge conflicts.

**Comment by glyph** [comment]
Since we can't update the branch and it seems unlikely that the original contributor is going to do the updating at this point, let's close this out. The functionality is desirable however, if anyone else is interested in making a new PR that does this.

**Pull request closed without merging**